### PR TITLE
Fix iree_atomic_load define

### PR DESCRIPTION
**Pull request by marbre**

### CODE REVIEW

**Comment by marbre** [comment]
The CMake build with GCC otherwise fails with
```
In file included from /home/mbrehler/repos/iree/iree/vm/module.h:21,
                 from /home/mbrehler/repos/iree/iree/vm/module.c:15:
/home/mbrehler/repos/iree/iree/base/atomics.h:75:15: error: expected identifier or ‘(’ before ‘{’ token
   75 | __extension__({
      |               ^
```

**Comment by benvanik** [comment]
🤦‍♂ thank you for catching this!

**Comment by benvanik** [comment]
(looks like there's a minor clang format issue)

File: iree/base/atomics.h [hunk 1]
```diff
@@ -71,13 +71,13 @@ typedef struct {
...
 typedef _Atomic __INTPTR_TYPE__ iree_atomic_intptr_t;
 #define IREE_ATOMIC_VAR_INIT(value) (value)
-#define iree_atomic_load(object) __atomic_load_ptr(object, __ATOMIC_SEQ_CST)
-__extension__({
-  __auto_type __atomic_load_ptr = (object);
-  __typeof__(*__atomic_load_ptr) __atomic_load_tmp;
-  __atomic_load(__atomic_load_ptr, &__atomic_load_tmp, (__ATOMIC_SEQ_CST));
-  __atomic_load_tmp;
-})
+#define iree_atomic_load(object) __atomic_load_ptr(object, __ATOMIC_SEQ_CST)  \
```
Comment (benvanik) [inline iree/base/atomics.h]: run clang-format to appease the github action gods :)

Comment (marbre) [reply iree/base/atomics.h]: Fixed :)

**Pull request closed without merging**